cpp/include/cudf/io/orc.hpp

```diff
@@ -178,7 +178,8 @@ class orc_reader_options {
   void set_stripes(std::vector<std::vector<size_type>> stripes)
   {
     CUDF_EXPECTS(stripes.empty() or (_skip_rows == 0), "Can't set stripes along with skip_rows");
-    CUDF_EXPECTS(stripes.empty() or (_num_rows == -1), "Can't set stripes along with num_rows");
+    CUDF_EXPECTS(stripes.empty() or not _num_rows.has_value(),
+                 "Can't set stripes along with num_rows");
```

Comment:
If stripes are empty, the code will throw in both exception handling. Do you mean `and` here?
```suggestion
    CUDF_EXPECTS(stripes.empty() and not _num_rows.has_value(),
                 "Can't set stripes along with num_rows");
```

Also, docs to be updated:
```cpp
  * @throw If no rows are skipped or stripes are empty
  * @throw If all rows need to be read but stripes are empty
```

Reply:
`or` is correct here. The failure case is when `not stripes.empty() and _num_rows.has_value()`. We need to expect the opposite here, which is the expression above.
I'll add `@throw` details to the docs👍